### PR TITLE
Add resultType extension method on Symbol

### PR DESCRIPTION
**Pull request by gabro**
This seemed like a useful utility so I extracted it from `ExplicitResultTypes`

### CODE REVIEW

File: scalafix-tests/unit/src/test/scala/scalafix/tests/DenotationOpsTest.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class DenotationOpsTest extends BaseSemanticTest("DenotationOpsTest") {
+
+  test("resultType") {
+    val source = docs.input.parse[Source].get
+    source.collect {
+      case t @ Pat.Var(Name("x")) =>
+        for {
+          symbol <- t.symbol
+          resultType <- symbol.resultType
+        } yield assert(resultType.structure == Type.Name("Boolean").structure)
```
Comment (gabro) [inline scalafix-tests/unit/src/test/scala/scalafix/tests/DenotationOpsTest.scala]: is there a smarter way to compare trees?

File: scalafix-core/shared/src/main/scala/scalafix/internal/util/DenotationOps.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package scalafix.internal.util
+
+import scala.meta._
+
+object DenotationOps {
+  val defaultDialect =
+    dialects.Scala212.copy(allowMethodTypes = true, allowTypeLambdas = true)
+
+  def resultType(
```
Comment (gabro) [inline scalafix-core/shared/src/main/scala/scalafix/internal/util/DenotationOps.scala]: the name is up to debate. `resultType` is general enough, but it's a bit of a stretch when referring to simple `val`/`var`

Comment (olafurpg) [reply scalafix-core/shared/src/main/scala/scalafix/internal/util/DenotationOps.scala]: Result type is the correct nomenclature from scalac 👍

File: scalafix-core/shared/src/main/scala/scalafix/syntax/package.scala [hunk 3]
```diff
@@ -30,6 +31,10 @@ package object syntax {
     @deprecated("Renamed to denotation", "0.5.0")
     def denotOpt: Option[Denotation] = denotation
     def denotation: Option[Denotation] = index.denotation(symbol)
+    def resultType(dialect: Dialect): Option[Type] =
+      denotation.flatMap(denot =>
+        DenotationOps.resultType(symbol, denot, dialect))
+    def resultType: Option[Type] = resultType(DenotationOps.defaultDialect)
```
Comment (olafurpg) [inline scalafix-core/shared/src/main/scala/scalafix/syntax/package.scala]: Can we only keep this overload and let advanced users with custom dialects call `DenotationOps.resultType(customDialect)`?

File: scalafix-tests/unit/src/test/scala/scalafix/tests/DenotationOpsTest.scala (hunk 1 above)
Comment (olafurpg) [reply scalafix-tests/unit/src/test/scala/scalafix/tests/DenotationOpsTest.scala]: There is! `import scala.meta.contrib._; q"a.b".isEqual(q"a.b.")`

File: scalafix-core/shared/src/main/scala/scalafix/syntax/package.scala (hunk 3 above)
Comment (gabro) [reply scalafix-core/shared/src/main/scala/scalafix/syntax/package.scala]: fine by me

File: scalafix-tests/unit/src/test/scala/scalafix/tests/DenotationOpsTest.scala (hunk 1 above)
Comment (gabro) [reply scalafix-tests/unit/src/test/scala/scalafix/tests/DenotationOpsTest.scala]: thanks! I'll fix it

**Comment by olafurpg** [comment]
We should filter out `scalafix.internal.*` from mima like here https://github.com/scalameta/scalafmt/blob/143bff64f24c4db6b53998a927d317ee0d874bcc/project/Mima.scala#L11

We need to upgrade Mima to 0.1.15 https://github.com/scalacenter/scalafix/blob/6b8cccd9438d8786742f02479865e16e434343fe/project/plugins.sbt#L17 which added support for regexes in filters

**Comment by gabro** [comment]
@olafurpg I've added the mima change in this PR, so that we can test it on the spot. Hope you don't mind

**Pull request merged**